src/cli/index.js

```diff
@@ -155,11 +155,16 @@ if (command.requireLockfile && !fs.existsSync(path.join(config.cwd, constants.LO
 //
 const run = (): Promise<void> => {
   invariant(command, 'missing command');
-  return command.run(config, reporter, commander, commander.args).then(() => {
+  return command.run(config, reporter, commander, commander.args).then(exitCode => {
     reporter.close();
     if (outputWrapper) {
       reporter.footer(false);
     }
```

Comment:
That would not work well, commands may resolve with non numeric values.
Also at line 337 in this file a rejection of run() promise causes process.exit(1) already.

Can you just throw an error in the outdated command?